src/lib/pubkey/pk_keys.h

```diff
@@ -111,6 +111,32 @@ class BOTAN_PUBLIC_API(3, 0) Asymmetric_Key {
        * key pair. For other key types it might throw Not_Implemented.
        */
       virtual std::unique_ptr<Private_Key> generate_another(RandomNumberGenerator& rng) const = 0;
+
+      /*
+      * Test the key values for consistency.
+      * @param rng rng to use
+      * @param strong whether to perform strong and lengthy version of the test
+      * @return true if the test is passed
+      */
+      virtual bool check_key(RandomNumberGenerator& rng, bool strong) const = 0;
+
+      // Declarations for internal library functions not covered by SemVer follow
+
+      /**
+      * Certain signatures schemes such as ECDSA have more than
+      * one element, and certain unfortunate protocols decided the
+      * thing to do was not concatenate them as normally done, but
+      * instead DER encode each of the elements as independent values.
+      *
+      * If this returns a value x then the signature is checked to
+      * be exactly 2*x bytes and split in half for DER encoding.
+      */
+      virtual std::optional<size_t> _signature_element_size() const { return {}; }
```

Comment:
```suggestion
      /**
      * Certain signatures schemes such as ECDSA have more than
      * one element, and certain unfortunate protocols decided the
      * thing to do was not concatenate them as normally done, but
      * instead DER encode each of the elements as independent values.
      *
      * If this returns a value x then the signature is checked to
      * be exactly 2*x bytes and split in half for DER encoding.
      */
      virtual std::optional<size_t> _signature_element_size_for_DER_encoding() const { return {}; }
```

... to make it clearer what the purpose of this function is. Especially when seeing it overridden in the relevant classes above, I wasn't sure what this was meant to do.

Reply:
Renamed